src/doing/utils.py

```diff
@@ -286,16 +286,16 @@ def guess_shell_encoding() -> str:
         return sys.stdout.encoding
 
 
-def verbose_shell():
+def verbose_shell() -> bool:
     """
     If shell commands should be printed.
 
     Users can define 'verbose_shell: True' in the .doing-cli-config.yml file.
     """
-    return get_config("verbose_shell", fallback=False)
+    return get_config("verbose_shell", fallback=False)  # type: ignore[no-any-return]
 
 
-def define_env(env):
+def define_env(env: Any) -> None:
```

Comment:
According to the documentation: https://mkdocs-macros-plugin.readthedocs.io/en/latest/macros/
_env is essentially an instance of a subclass of the MkDocs' [BasePlugin class](https://www.mkdocs.org/user-guide/plugins/#baseplugin), with some additional properties. Whatever you find in the BasePlugin class, you will find in the the env object._

However, `BasePlugin` also doesn't have a `macro` attribute or method, so it doesn't make sense to annotate that it a such. I opted to add the ignore as that is cleaner.